python/pyspark/sql/types.py

```diff
@@ -1702,9 +1791,16 @@ def _parse_datatype_json_string(json_string: str) -> DataType:
     return _parse_datatype_json_value(json.loads(json_string))
 
 
-def _parse_datatype_json_value(json_value: Union[dict, str]) -> DataType:
+def _parse_datatype_json_value(
+    json_value: Union[dict, str],
+    fieldPath: str = "",
+    collationsMap: Optional[Dict[str, str]] = None,
+) -> DataType:
     if not isinstance(json_value, dict):
         if json_value in _all_atomic_types.keys():
+            if collationsMap is not None and fieldPath in collationsMap:
+                collationName = collationsMap[fieldPath].split(".")[1]
```

Comment:
So we are using only the name? What about the rest of the parts?

Reply:
In spark we only use name to identify collations and we don't need the provider at all. That is why I asked if provider could be optional in the delta protocol but the issue would be parsing the identifier in that case

https://github.com/delta-io/delta/pull/3068#discussion_r1593742078

Reply:
How about an error here if the provider is not Spark or ICU?

Reply:
sure!